xcm/xcm-executor/src/assets.rs

```diff
@@ -190,6 +250,98 @@ impl Assets {
 			match asset {
 				MultiAsset::None => (),
 				MultiAsset::All => return self.swapped(Assets::default()),
+				x @ MultiAsset::AllFungible | x @ MultiAsset::AllNonFungible => {
+					let all_fungible = Assets {
+						fungible: self.fungible.clone(),
+						non_fungible: Default::default(),
+					};
+					let all_non_fungible = Assets {
+						fungible: Default::default(),
+						non_fungible: self.non_fungible.clone()
+					};
+					match x {
+						MultiAsset::AllFungible => {
+							self.swapped(all_non_fungible);
+							return all_fungible
+						},
+						MultiAsset::AllNonFungible => {
+							self.swapped(all_fungible);
+							return all_non_fungible
+						},
+						_ => unreachable!(),
+					}
+
+				},
+				MultiAsset::AllAbstractFungible { id } => {
+					let all_abstract_fungible: Vec<(AssetId, u128)> = self.fungible.clone()
+						.into_iter()
+						.filter(|(iden, _)| match iden {
+							AssetId::Abstract(iden) => id == *iden,
+							_ => false,
+						}).collect();
+
+					for (id, amount) in all_abstract_fungible {
+						let maybe_value = self.fungible.get(&id);
+						if let Some(&e) = maybe_value {
+							if e > amount {
+								self.fungible.insert(id.clone(), e - amount);
+								result.saturating_subsume_fungible(id, amount);
+							} else {
+								self.fungible.remove(&id);
+								result.saturating_subsume_fungible(id, e.clone());
+							}
+						}
+					}
```

Comment:
I am not confident that this is the optimal way to do this. Would love a close review (as this logic is duplicated 4 times below)